Enable CLI parameter specification via env vars

FabijanC opened the pull request:
## Usage related changes

- Closes #482
- Closes #475
- More info in the updated .md documentation files
- Improves `#[arg(value_name = ...)]` of some CLI parameters.

## Development related changes

- Enable the `env` feature of the `clap` dependency in `Cargo.toml`.
- Introduces `serial_test` dev dependency because env-mainpulating tests cannot be run in parallel.
  - Apparently `env::remove_var` [should be unsafe but isn't](https://doc.rust-lang.org/std/env/fn.remove_var.html).

## Checklist:

- [x] Checked out the [contribution guidelines](CONTRIBUTING.md)
- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] No unused dependencies - `./scripts/check_unused_deps.sh`
- [x] No spelling errors - `./scripts/check_spelling.sh`
- [x] Performed code self-review
- [x] Rebased to the latest commit of the target branch (or merged it into my branch)
- [x] Updated the docs if needed - `./website/README.md`
- [x] Linked the [issues](https://github.com/0xSpaceShard/starknet-devnet-rs/issues) resolvable by this PR - [linking info](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [x] Updated the tests if needed; all passing - [execution info](https://github.com/0xSpaceShard/starknet-devnet-rs/?tab=readme-ov-file#test-execution)

marioiordanov reviewed (commented):
What will happen if cli args and env variables are passed

FabijanC commented:
> What will happen if cli args and env variables are passed

I thought I added this info, thanks for reminding me. Updated the docs in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/483/commits/e268e43c7bfbacee77f9040e1cc1c75ede011f5a
